File: backend-java/src/main/java/com/taru/taskmanager/models/StatusTasks.java
package com.taru.taskmanager.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "status_tasks")
public class StatusTasks {

    @EmbeddedId
    private StatusTasksId id;
    @ManyToOne
    @MapsId("taskId")
    @JoinColumn(name = "task_id")
    private Task task;
    @ManyToOne
    @MapsId("statusId")
    @JoinColumn(name = "status_id")
    private Status status;
}

File: backend-csharp/TaskManager.Notification/Entities/StatusTasks.cs
using System;
using System.Collections.Generic;

namespace TaskManager.Notification.Entities;

public partial class StatusTasks
{
    public int StatusId { get; set; }

    public int TaskId { get; set; }

    public virtual Status Status { get; set; } = null!;

    public virtual Tasks Task { get; set; } = null!;
}
